test-src/epfl/test14-scratch/TestQueries.scala

```diff
@@ -797,11 +798,18 @@ trait StagedExp extends Staged with ScalaOpsPkgExp with BooleanOpsExpOpt with St
             if P then (if Q then R) --> if (P && Q) then R
         if P then (for x in Q do R) --> for x in Q do (if P then R)
 */
+    // NOTE(trans): we need to use List[T]() instead of List().
```

Comment:
This is a really strangely written test. The unapply methods have type parameters, which I first assumed wouldn't do anything because of type erasure, but it turns out they're required but the compiler can't make the jump that type B from the unapply method is equal to type T of the input? Why did this work before?